### PR TITLE
HW12

### PR DESCRIPTION
**Pull request by Artsapollo**
image

### CODE REVIEW

File: src/test/java/HW11/FruitShopMockSpyTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class FruitShopMockSpyTest {
+    private List<Vitamins> vitamins = Arrays.asList(Vitamins.A);
+    private Fruit apple = new Fruit(APPLE, 10, LocalDate.of(2019, 1, 1), 10, vitamins);
+    @Spy
+    @InjectMocks
+    private FruitShop fruitShop;
```
Comment (mkharkhu) [inline src/test/java/HW11/FruitShopMockSpyTest.java]: Is it used somewhere?

File: src/test/java/HW11/FruitShopMockSpyTest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    @Test
+    public void mockTestAllFruitsOfFruitType() {
+        when(appleMock.getFruitType()).thenReturn(FruitType.APPLE);
+        FruitType expected = FruitType.APPLE;
+        FruitType actual = appleMock.getFruitType();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void mockTestDaysToLive() {
```
Comment (mkharkhu) [inline src/test/java/HW11/FruitShopMockSpyTest.java]: a little confusing with this test. you make mock on a method and check it?

File: src/test/java/HW11/FruitShopMockSpyTest.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        System.out.println("After");
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        System.out.println("After class");
+    }
+
+    @Test
+    public void mockTestAllFruitsOfFruitType() {
```
Comment (SergeyPasko) [inline src/test/java/HW11/FruitShopMockSpyTest.java]: bad names for methods

File: src/test/java/HW11/FruitShopMockSpyTest.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    private List<Vitamins> vitamins = Arrays.asList(Vitamins.A);
+    private Fruit apple = new Fruit(APPLE, 10, LocalDate.of(2019, 1, 1), 10, vitamins);
+    @Spy
+    @InjectMocks
+    private FruitShop fruitShop;
+    @Mock
+    private Fruit appleMock;
+
+    @BeforeClass
+    public static void beforeClass() {
```
Comment (SergeyPasko) [inline src/test/java/HW11/FruitShopMockSpyTest.java]: Why you use this is?

Comment (SergeyPasko) [reply src/test/java/HW11/FruitShopMockSpyTest.java]: Other below also

File: src/test/java/HW11/FruitShopMockSpyTest.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        FruitType expected = FruitType.APPLE;
+        FruitType actual = appleMock.getFruitType();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void mockTestDaysToLive() {
+        when(appleMock.getDayToLive()).thenReturn(10);
+        int expected = 10;
+        int actual = appleMock.getDayToLive();
```
Comment (SergeyPasko) [inline src/test/java/HW11/FruitShopMockSpyTest.java]: you test there is that your mock is work
you don't test daysToLive method

**Comment by Artsapollo** [comment]
Okay, so, I have read your comments. Added log4J2.xml file and made some magic with tests. Decided to leave these tests which are showing how mock and spy work.

**Pull request merged**